synapseclient/__main__.py

```diff
@@ -534,21 +540,27 @@ def _replace_existing_config(path, auth_section):
     return new_config_text
 
 
-def _generate_new_config(auth_section):
+def _generate_new_config(auth_section, profile_name):
     # insert the auth section into the default config template file
 
     script_dir = os.path.dirname(os.path.realpath(__file__))
 
     with open(os.path.join(script_dir, ".synapseConfig"), "r") as config_o:
         config_text = config_o.read()
 
-    # Replace text in configuration
-    new_config_text = re.sub(
-        r"#\[authentication\].*<authtoken>",
-        auth_section,
-        config_text,
-        flags=re.MULTILINE | re.DOTALL,
-    )
+    if not profile_name:
+        # Replace text in configuration
+        new_config_text = re.sub(
+            r"#\[default\].*<authtoken>",
+            auth_section,
+            config_text,
+            flags=re.MULTILINE | re.DOTALL,
+        )
+
+    if profile_name:
```

Comment:
NIT: I think this would be a little more readable if you just used `else` for the second condition since these are mutually exclusive paths in the code. Usually when I see this pattern I assume that it's possible for both code blocks to be executed.